### PR TITLE
FILTER: ApplyTransformationToGeometry

### PR DESCRIPTION
**Pull request by imikejackson**

### CODE REVIEW

**Review by mmarineBlueQuartz — APPROVED** [review]

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]
There's an extra set of `ApplyTransformationToGeometry` files that are empty.

**Review by mmarineBlueQuartz — DISMISSED** [review]

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+
+  // Create default Parameters for the filter.
+  args.insertOrAssign(StlFileReaderFilter::k_StlFilePath_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  args.insertOrAssign(StlFileReaderFilter::k_GeometryDataPath_Key, std::make_any<DataPath>(triangleGeomDataPath));
+  args.insertOrAssign(StlFileReaderFilter::k_FaceGroupDataPath_Key, std::make_any<DataPath>(triangleFaceDataGroupDataPath));
+  args.insertOrAssign(StlFileReaderFilter::k_FaceNormalsDataPath_Key, std::make_any<DataPath>(normalsDataPath));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(dataGraph, args);
+  REQUIRE(preflightResult.outputActions.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:96]:
```suggestion
  COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+
+  // Execute the filter and check the result
+  auto executeResult = filter.execute(dataGraph, args);
+  REQUIRE(executeResult.result.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:100]:
```suggestion
  COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+    // Create default Parameters for the filter.
+    DataPath triangleAreasDataPath = geometryPath.createChildPath(triangleFaceDataGroupName).createChildPath(triangleAreasName);
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_GeometryToTransform_Key, std::make_any<DataPath>(geometryPath));
+
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_TransformType_Key, std::make_any<complex::ChoicesParameter::ValueType>(4));
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_Translation_Key, std::make_any<complex::VectorFloat32Parameter::ValueType>({100.0F, 100.0F, 100.0F}));
+
+    // Preflight the filter and check result
+    auto preflightResult = filter.preflight(dataGraph, args);
+    REQUIRE(preflightResult.outputActions.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:127]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+
+    // Execute the filter and check the result
+    auto executeResult = filter.execute(dataGraph, args);
+    REQUIRE(executeResult.result.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:131]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+    // Create default Parameters for the filter.
+    DataPath triangleAreasDataPath = geometryPath.createChildPath(triangleFaceDataGroupName).createChildPath(triangleAreasName);
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_GeometryToTransform_Key, std::make_any<DataPath>(geometryPath));
+
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_TransformType_Key, std::make_any<complex::ChoicesParameter::ValueType>(3));
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_RotationAxisAngle_Key, std::make_any<complex::VectorFloat32Parameter::ValueType>({0.0F, 0.0F, 1.0F, 45.0F}));
+
+    // Preflight the filter and check result
+    auto preflightResult = filter.preflight(dataGraph, args);
+    REQUIRE(preflightResult.outputActions.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:180]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+
+    // Execute the filter and check the result
+    auto executeResult = filter.execute(dataGraph, args);
+    REQUIRE(executeResult.result.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:184]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+    // Create default Parameters for the filter.
+    DataPath triangleAreasDataPath = geometryPath.createChildPath(triangleFaceDataGroupName).createChildPath(triangleAreasName);
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_GeometryToTransform_Key, std::make_any<DataPath>(geometryPath));
+
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_TransformType_Key, std::make_any<complex::ChoicesParameter::ValueType>(5));
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_Scale_Key, std::make_any<complex::VectorFloat32Parameter::ValueType>({2.0F, 3.0F, 4.0F}));
+
+    // Preflight the filter and check result
+    auto preflightResult = filter.preflight(dataGraph, args);
+    REQUIRE(preflightResult.outputActions.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:233]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_TransformType_Key, std::make_any<complex::ChoicesParameter::ValueType>(2));
+
+    // This should reflect the geometry across the x-axis.
+    DynamicTableParameter::ValueType dynamicTable{{{-1.0, 0, 0, 0}, {0, 1.0, 0, 0}, {0, 0, 1.0, 0}, {0, 0, 0, 1.0}}, {"R0", "R1", "R2", "R3"}, {"C0", "C1", "C2", "C3"}};
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_ManualTransformationMatrix_Key, std::make_any<complex::DynamicTableParameter::ValueType>(dynamicTable));
+
+    // Preflight the filter and check result
+    auto preflightResult = filter.preflight(dataGraph, args);
+    REQUIRE(preflightResult.outputActions.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:289]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(preflightResult.outputActions);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 9]
```diff
@@ -0,0 +1,398 @@
...
+
+    // Execute the filter and check the result
+    auto executeResult = filter.execute(dataGraph, args);
+    REQUIRE(executeResult.result.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:293]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```

File: src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp [hunk 10]
```diff
@@ -0,0 +1,398 @@
...
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_TransformType_Key, std::make_any<complex::ChoicesParameter::ValueType>(1));
+    args.insertOrAssign(ApplyTransformationToGeometryFilter::k_ComputedTransformationMatrix_Key, std::make_any<DataPath>({precomputedName}));
+
+    // Preflight the filter and check result
+    auto preflightResult = filter.preflight(dataGraph, args);
+    REQUIRE(preflightResult.outputActions.valid());
+
+    // Execute the filter and check the result
+    auto executeResult = filter.execute(dataGraph, args);
+    REQUIRE(executeResult.result.valid());
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/test/ApplyTransformationToGeometryFilterTest.cpp:371]:
```suggestion
    COMPLEX_RESULT_REQUIRE_VALID(executeResult.result);
```

**Review by JDuffeyBQ — DISMISSED** [review]

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ApplyTransformationToGeometry.cpp [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+#include <cstdint>
+#include <map>
+
+using namespace complex;
+
+namespace ApplyTransformationProgress
+{
+static size_t s_InstanceIndex = 0;
+static std::map<size_t, int64_t> s_ProgressValues;
+static std::map<size_t, int64_t> s_LastProgressInt;
```
[the remark this answers is not in the document]
Comment (imikejackson) [reply src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ApplyTransformationToGeometry.cpp:26]: It works just fine the way it is?

**Review by mmarineBlueQuartz — APPROVED** [review]

**Pull request merged**